pystiche/papers/ulyanov_et_al_2016/data.py

```diff
@@ -0,0 +1,269 @@
+from typing import Optional, Sized
+from urllib.parse import urljoin
+import torch
+from torch.utils.data import Dataset, Sampler, DataLoader
+from pystiche.data import (
+    DownloadableImage,
+    DownloadableImageCollection,
+    CreativeCommonsLicense,
+    PublicDomainLicense,
+    ImageFolderDataset,
+    FiniteCycleBatchSampler,
+)
+from pystiche.image import extract_num_channels
+from pystiche.image.transforms import Transform, Resize, ComposedTransform, CenterCrop
+from pystiche.image.transforms.functional import grayscale_to_fakegrayscale
+
+
+__all__ = [
+    "ulyanov_et_al_2016_content_transform",
+    "ulyanov_et_al_2016_style_transform",
+    "ulyanov_et_al_2016_images",
+    "ulyanov_et_al_2016_dataset",
+    "ulyanov_et_al_2016_batch_sampler",
+    "ulyanov_et_al_2016_image_loader",
+]
+
+
+def ulyanov_et_al_2016_content_transform(
+    edge_size: int = 256, impl_params: bool = True, instance_norm: bool = True,
+) -> ComposedTransform:
+    class OptionalGrayscaleToFakegrayscale(Transform):
+        def forward(self, input_image: torch.Tensor) -> torch.Tensor:
+            is_grayscale = extract_num_channels(input_image) == 1
+            if is_grayscale:
+                return grayscale_to_fakegrayscale(input_image)
+            else:
+                return input_image
+
+    transforms = []
+    if impl_params:
+        if instance_norm:
+            transforms.append(CenterCrop(edge_size))
+        else:
+            transforms.append(
+                Resize((edge_size, edge_size), interpolation_mode="bilinear")
+            )
+    else:
+        transforms.append(CenterCrop(edge_size))
```

Comment:
Shouldn't be a random crop here somewhere?

Reply:
Yes, a random crop should be performed for `impl_params and instance_norm`. `RandomCrop` still has to be implemented as a subclass of `Transform`. Probably with a set random seed to make our implementation reproducible?

Reply:
`torchvision` already implements a [`RandomCrop` transform](https://github.com/pytorch/vision/blob/3c254fb7af5f8af252c24e89949c54a3461ff0be/torchvision/transforms/transforms.py#L399). Maybe we can use it as a starting point. I'll take that up and get back to you when its done.

Reply:
It would be great if you did.

Reply:
It is done. Pull to get the latest version.